File: Chat/PreserveUserTextMessageHistory/PreserveUserTextMessageHistoryVB/CustomVisualItem.vb
Imports Telerik.WinControls
Imports Telerik.WinControls.UI

Public Class CustomVisualItem
    Inherits RadListVisualItem

    Private myLightVisualElement As LightVisualElement
    Private font1 As FontFamily

    Protected Overrides Sub CreateChildElements()
        font1 = ThemeResolutionService.GetCustomFont(ThemeResolutionService.TelerikWebUIFontName)
        MyBase.CreateChildElements()
        Me.myLightVisualElement = New LightVisualElement()
        Me.myLightVisualElement.CustomFont = font1.Name
        Me.myLightVisualElement.Visibility = ElementVisibility.Hidden
        Me.myLightVisualElement.Text = TelerikWebUIFont.GlyphEmail
        Me.myLightVisualElement.TextAlignment = ContentAlignment.MiddleRight
        Me.Children.Add(Me.myLightVisualElement)
    End Sub

    Public Property MessageNotification As ElementVisibility
        Get
            Return Me.myLightVisualElement.Visibility
        End Get
        Set(ByVal value As ElementVisibility)
            Me.myLightVisualElement.Visibility = value
        End Set
    End Property

    Protected Overrides ReadOnly Property ThemeEffectiveType As Type
        Get
            Return GetType(RadListVisualItem)
        End Get
    End Property
End Class

File: Chat/PreserveUserTextMessageHistory/PreserveUserTextMessageHistoryCSharp/CustomVisualItem.cs
using System;
using System.Drawing;
using Telerik.WinControls;
using Telerik.WinControls.UI;

namespace PreserveUserTextMessageHistory
{
    public class CustomVisualItem : RadListVisualItem
    {
        private LightVisualElement lve = null;
        FontFamily font1 = ThemeResolutionService.GetCustomFont(ThemeResolutionService.TelerikWebUIFontName);

        protected override void CreateChildElements()
        {
            base.CreateChildElements();
            this.lve = new LightVisualElement();
            this.lve.CustomFont = font1.Name;
            lve.Visibility = ElementVisibility.Hidden;
            lve.Text = TelerikWebUIFont.GlyphEmail;
            lve.TextAlignment = ContentAlignment.MiddleRight;
            this.Children.Add(lve);
        }

        public ElementVisibility MessageNotification
        {
            get { return lve.Visibility; }
            set { lve.Visibility = value; }
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(RadListVisualItem);
            }
        }
    }
}
